cpp/arcticdb/version/test/test_version_map.cpp

```diff
@@ -902,19 +907,11 @@ TEST(VersionMap, CacheInvalidationWithTombstoneAllAfterLoad) {
         // When - we delete all versions without reloading
         version_map->write_tombstone_all_key_internal(store, tombstone_key, entry);
 
-        if (is_loaded_to_0) {
-            // If we have loaded everything we should not invalidate cache
-            ASSERT_TRUE(version_map->has_cached_entry(id, LoadStrategy{LoadType::LATEST, LoadObjective::UNDELETED_ONLY}));
-            ASSERT_TRUE(version_map->has_cached_entry(id, LoadStrategy{LoadType::FROM_TIME, LoadObjective::UNDELETED_ONLY, static_cast<timestamp>(2)}));
-            ASSERT_TRUE(version_map->has_cached_entry(id, LoadStrategy{LoadType::FROM_TIME, LoadObjective::UNDELETED_ONLY, static_cast<timestamp>(1)}));
-            ASSERT_TRUE(version_map->has_cached_entry(id, LoadStrategy{LoadType::FROM_TIME, LoadObjective::UNDELETED_ONLY, static_cast<timestamp>(0)}));
-        }
-        else {
-            // If we haven't loaded everything we should invalidate cached undeleted checks
-            ASSERT_FALSE(version_map->has_cached_entry(id, LoadStrategy{LoadType::LATEST, LoadObjective::UNDELETED_ONLY}));
-            ASSERT_FALSE(version_map->has_cached_entry(id, LoadStrategy{LoadType::FROM_TIME, LoadObjective::UNDELETED_ONLY, static_cast<timestamp>(1)}));
-            ASSERT_FALSE(version_map->has_cached_entry(id, LoadStrategy{LoadType::FROM_TIME, LoadObjective::UNDELETED_ONLY, static_cast<timestamp>(0)}));
-        }
+        // Tombstone All should not invalidate cache as it deletes everything so all undeleted versions have been loaded.
```

Comment:
This is something I caught when testing. Previously writing tombstone_all would cause any subsequent UNDELETED_ONLY lookups in cache to return false. However, I think this was wrong because writing tombstone_all deletes all previous versions so it is impossible for an undeleted version to exist in the version chain and not in the cache. Thus all undeleted versions have been loaded in cache.

Reply:
Add tests where the tombstone_all version ID is not the latest version?